cpp/dolfinx/mesh/graphbuild.cpp

```diff
@@ -486,16 +504,15 @@ mesh::build_local_dual_graph(CellType celltype,
 }
 //-----------------------------------------------------------------------------
 graph::AdjacencyList<std::int64_t>
-mesh::build_dual_graph(MPI_Comm comm, CellType celltype,
-                       const graph::AdjacencyList<std::int64_t>& cells)
+mesh::build_dual_graph(MPI_Comm comm, const std::vector<CellType>& celltypes,
```

Comment:
```suggestion
mesh::build_dual_graph(MPI_Comm comm, std::span<const CellType> celltypes,
```
Wouldn't a span be more natural in this context, as we try avoiding using std::vector in the code-base in general? 
I get that you can't do it for `cells` as it is closer to a list of adjacency lists.